tethys_dataset_services/engines/geoserver_engine.py

```diff
@@ -1238,84 +1374,117 @@ def add_table_to_postgis_store(self, store_id, table, debug=False):
         self._handle_debug(response_dict, debug)
         return response_dict
 
-    def create_sql_view(self, feature_type_name, postgis_store_id, sql, geometry_column, geometry_type,
-                        geometry_srid=4326, default_style_id=None, key_column=None, parameters=None, debug=False):
+    def create_sql_view_layer(self, layer_id, postgis_store_name, geometry_type, srid, sql, default_style, 
+                              geometry_name='geometry', other_styles=None, parameters=None, debug=False):
         """
-        Create a new feature type configured as an SQL view.
+        Direct call to GeoServer REST API to create SQL View feature types and layers.
 
         Args:
-          feature_type_name (string): Name of the feature type and layer to be created.
-          postgis_store_id (string): Identifier of existing postgis store with tables that will be queried by the sql view. Can be a store name or a workspace-name combination (e.g.: "name" or "workspace:name").
-          sql (string): SQL that will be used to construct the sql view / virtual table.
-          geometry_column (string): Name of the geometry column.
-          geometry_type (string): Type of the geometry column (e.g. "Point", "LineString", "Polygon").
-          geometry_srid (string, optional): EPSG spatial reference id of the geometry column. Defaults to 4326.
-          default_style_id (string, optional): Identifier of a style to assign as the default style. Can be a style name or a workspace-name combination (e.g.: "name" or "workspace:name").
-          key_column (string, optional): The name of the key column.
-          parameters (iterable, optional): A list/tuple of tuple-triplets representing parameters in the form (name, default, regex_validation), (e.g.: (('variable', 'pressure', '^[\w]+$'), ('simtime', '0:00:00', '^[\w\:]+$'))
-          debug (bool, optional): Pretty print the response dictionary to the console for debugging. Defaults to False.
-
-        Returns:
-          (dict): Response dictionary
+            layer_id (string): Identifier of the sql view layer to create. Can be a name or a workspace-name combination (e.g.: "name" or "workspace:name").
+            postgis_store_name (string): Identifier of existing postgis store with tables that will be queried by the sql view. (assumption: the datastore belongs to the workspace).
```

Comment:
The workspace should be associated with the store, not the layer.

Change `postgis_store_name` to `store_id` and extract the workspace from that. Update the doc strings too.

Also, change `layer_id` to `layer_name`. Update the doc strings too.